src/wallet/wallet.cpp

```diff
@@ -102,19 +102,11 @@ static void DeleteWallet(CWallet* wallet)
 // its calling thread, so that only that thread waits for the unload.
 static void ReleaseWallet(CWallet* wallet)
 {
-    bool unloading{false};
-    {
-        LOCK(g_wallet_release_mutex);
-        unloading = g_unloading_wallet_set.erase(wallet) == 1;
-    }
-    if (unloading) {
-        // Delegate wallet deletion to UnloadWallet caller, just wake it up.
-        g_wallet_release_cv.notify_all();
-    } else {
-        // UnloadWallet was not called, just delete the wallet since there's no
-        // more shared pointers to the wallet.
-        DeleteWallet(wallet);
-    }
+    DeleteWallet(wallet);
```

Comment:
Would suggest inlining `DeleteWallet` here. This function is only called from this one place, and I can't think of any other place where it would be safe to call.

Reply:
Good point.